lib/extensions/permissions.js

```diff
@@ -215,16 +220,84 @@ async function setAccess (bundleId, permissionsMapping) {
     const permissionsArg = _.toPairs(wixPermissions)
       .map((x) => `${x[0]}=${formatStatus(x[1])}`)
       .join(',');
-    await execWix.bind(this)([
+    const execWixFn = async () => await execWix.bind(this)([
       '--byId', this.udid,
       '--bundle', bundleId,
       '--setPermissions', permissionsArg,
     ]);
+    const shouldWaitForSystemReadiness = !_.isEmpty(
+      _.intersection(SERVICES_NEED_SPRINGBOARD_RESTART, _.keys(wixPermissions))
+    );
+    if (shouldWaitForSystemReadiness) {
+      const [didTimeout] = await runAndWaitForSystemReadiness.bind(this)(
+        execWixFn, SYSTEM_SERVICE_RESTART_TIMEOUT_MS
+      );
+      if (didTimeout) {
+        this.log.warn(
+          `The Springboard service and apps restarted by the springboard process restart did not restart ` +
```

Comment:
The required system services did not restart after ...

Reply:
done